assets/js/payplug-apple-pay-cart.js

```diff
@@ -230,36 +227,39 @@
 		CancelOrder: function (message) {
 			session.oncancel = event => {
 				$('apple-pay-button').removeClass("isDisabled")
-				jQuery.post({
-					url : apple_pay_params.ajax_url_applepay_cancel_order,
-					data : {
-						'order_id' : session.order_id,
-						'payment_id' : session.payment_id
-					}
-				}).done(function (response) {
-					if (message.length > 0) {
-						apple_pay.showError(message)
-					} else {
-						apple_pay.showError(response.data.message)
-					}
-				})
+				apple_pay.cancel_order_request(session.order_id, session.payment_id);
 
 			}
 		},
-
-		showError: function (message) {
+		handle_process_error: function (order_id = null) {
+			$('apple-pay-button').removeClass("isDisabled")
+			apple_pay.cancel_order_request(order_id, null);
+		},
+		cancel_order_request: function(order_id, payment_id = null){
+			jQuery.post({
+				url : apple_pay_params.ajax_url_applepay_cancel_order,
+				data : {
+					'order_id' : order_id,
+					'payment_id' : payment_id
+				}
+			}).done(function (response) {
+				apple_pay.showError(response.data.message, "info");
+			})
```

Comment:
Just checking if this post has an status code 4xx or 5xx, wont be treated by the done, we'll need to add a 
```javascript
.done(function (response) {
				apple_pay.showError(response.data.message, "info");
			}). fail(function(error){ 
// treat the error
})
```

Reply:
it is hard for that request to send a 4xx or 5xx, but to be safe yes we can add it

Reply:
the goal for this US was precisely treat this errors, with the usage of `wp_send_json_error` on the output of the `Exception` you'll always should get a json response. Of course for unplanned Exceptions, this will generate a 500 error and we'll have the same problem has ever (because there's no order_id you cannot rebuilt the cart, cancel order, ...).